### PR TITLE
setup Sphinx

### PR DESCRIPTION
**Pull request by sangyx**
I set up Sphinx and added a simple quickstart.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        $ git clone https://github.com/Quansight-Labs/udiff.git
+        $ cd udiff
+
+#.  Install ``udiff`` on the command line, enter::
+
+        $ python setup.py install --user
+
+#.  Use ``udiff`` in your code. A simple example::
+
+         import uarray as ua
```
Comment (hameerabbasi) [inline docs/index.rst]: We can't test these examples. 😦 And the docs may go out of date. Can you add a docstring to the `udiff` module and use autodoc?

Docstring example: https://github.com/Quansight-Labs/unumpy/blob/master/unumpy/__init__.py
Autodoc: https://github.com/Quansight-Labs/unumpy/tree/master/docs/generated

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
One small comment.

File: docs/index.rst (hunk 1 above)
Comment (sangyx) [reply docs/index.rst]: Fine. I'll try.

Comment (sangyx) [reply docs/index.rst]: Hi, I made some changes, could you please take a look?

File: src/udiff/__init__.py [hunk 2]
```diff
@@ -1,3 +1,39 @@
...
+#.  Use Git to clone the :obj:`udiff` repository::
+
+        $ git clone https://github.com/Quansight-Labs/udiff.git
+        $ cd udiff
+
+#.  Install :obj:`udiff` on the command line, enter::
+
+        $ python setup.py install --user
+
+#.  Use :obj:`udiff` in your code. A simple example::
```
Comment (hameerabbasi) [inline src/udiff/__init__.py]:
```suggestion
#.  Use :obj:`udiff` in your code. A simple example:
```

File: src/udiff/__init__.py [hunk 3]
```diff
@@ -1,3 +1,39 @@
...
+        >>> from unumpy import numpy_backend
+        >>> from numpy import allclose
+
+        >>> with ua.set_backend(numpy_backend), ua.set_backend(udiff, coerce=True):
+        ...    x = np.reshape(np.arange(25), (5, 5))
+        ...    x.var = udiff.Variable('x')
+        ...    y = np.exp(2 * x)
+        ...    y_d = 2 * y
+        ...    print(allclose(y.diffs[x].arr, y_d.arr))
+        True
```
Comment (hameerabbasi) [inline src/udiff/__init__.py]: No indent needed here.

File: src/udiff/__init__.py [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+
+Getting started is easy:
+
+.. note::
+    :obj:`udiff` has not been published on PyPI. You have to install it from source code now.
+
+#.  Use Git to clone the :obj:`udiff` repository::
+
+        $ git clone https://github.com/Quansight-Labs/udiff.git
+        $ cd udiff
```
Comment (hameerabbasi) [inline src/udiff/__init__.py]: I would use the `..code: bash` directive.

File: src/udiff/__init__.py [hunk 5]
```diff
@@ -1,3 +1,39 @@
...
+
+#.  Install :obj:`udiff` on the command line, enter::
+
+        $ python setup.py install --user
```
Comment (hameerabbasi) [inline src/udiff/__init__.py]: `..code: bash`

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
A couple of small changes. Sorry, I reviewed before but forgot to submit the review.

File: src/udiff/__init__.py (hunk 3 above)
Comment (sangyx) [reply src/udiff/__init__.py]: There are two indents. Should I delete one or two?

Comment (hameerabbasi) [reply src/udiff/__init__.py]: All indents for Python blocks.

**Comment by hameerabbasi** [comment]
I guess the remaining errors can be cleaned up once we have CI.

**Pull request merged**